src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerOfflineDocumentVerified.java

```diff
@@ -178,24 +183,35 @@ public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(CaseDetails<C
         log.info("{} about to submit callback invoked for Case Id: {}", CASEWORKER_OFFLINE_DOCUMENT_VERIFIED, details.getId());
         var caseData = details.getData();
 
-        if (AOS_D10.equals(caseData.getDocuments().getTypeOfDocumentAttached())) {
+        if (AOS_D10.equals(caseData.getDocuments().getTypeOfDocumentAttached())
+            || D10.equals(caseData.getDocuments().getScannedSubtypeReceived())) {
 
-            reclassifyScannedDocumentToChosenDocumentType(caseData, RESPONDENT_ANSWERS);
+            if (isEmpty(caseData.getDocuments().getScannedSubtypeReceived())) {
+                reclassifyScannedDocumentToChosenDocumentType(caseData, RESPONDENT_ANSWERS);
+            }
             // setting the status as drafted as AOS answers has been received and is being classified by caseworker
             details.setState(AosDrafted);
 
             final CaseDetails<CaseData, State> response = submitAosService.submitOfflineAos(details);
             response.getData().getApplicant2().setOffline(YES);
             response.getData().getAcknowledgementOfService().setStatementOfTruth(YES);
+            response.getData().getDocuments().setScannedSubtypeReceived(null);
 
             return AboutToStartOrSubmitResponse.<CaseData, State>builder()
                 .data(response.getData())
                 .state(response.getState())
                 .build();
 
-        } else if (CO_D84.equals(caseData.getDocuments().getTypeOfDocumentAttached())) {
+        } else if (CO_D84.equals(caseData.getDocuments().getTypeOfDocumentAttached())
+            || D84.equals(caseData.getDocuments().getScannedSubtypeReceived())) {
+
+            if (isEmpty(caseData.getDocuments().getScannedSubtypeReceived())) {
+                reclassifyScannedDocumentToChosenDocumentType(caseData, CONDITIONAL_ORDER_APPLICATION);
+            }
 
-            reclassifyScannedDocumentToChosenDocumentType(caseData, CONDITIONAL_ORDER_APPLICATION);
+            if (!SWITCH_TO_SOLE.equals(caseData.getConditionalOrder().getD84ApplicationType())) {
+                caseData.getDocuments().setScannedSubtypeReceived(null);
```

Comment:
if we set this to null then we won't be able to search it from search filter ?

Reply:
This is because we use the value to filter D36, D10 and D84 that need to be picked up by Caseworkers to action, once actioned we don't want these cases to be filtered anymore